warc_metadata_sidecar.py

```diff
@@ -0,0 +1,207 @@
+import argparse
+import io
+import logging
+import os
+import re
+import time
+from datetime import timedelta
+
+import chardet
+import pycld2 as cld2
+from fido.fido import Fido
+from warcio.archiveiterator import ArchiveIterator
+from warcio.recordloader import ArcWarcRecord
+from warcio.warcwriter import WARCWriter
+
+
+class ExtendFido(Fido):
+    """A class that extends Fido to override some methods."""
+    def blocking_read(self, file, bytes_to_read):
```

Comment:
So we remember, it would be good to include a docstring that indicates why we are overriding it.